spec/resources/groups_spec.rb

```diff
@@ -0,0 +1,59 @@
+require 'spec_helper'
+
+describe TableauApi::Resources::Groups do
+  include_context 'tableau client'
+
+  describe '#create' do
+    # http://onlinehelp.tableau.com/v9.0/api/rest_api/en-us/help.htm#REST/rest_api_ref.htm#Create_Group
+    it 'fails with a bad site_role' do
+      VCR.use_cassette('groups') do
+        expect { client.groups.create(name: 'test', default_site_role: 'foo') }.to raise_error('invalid default_site_role')
+      end
+    end
+
+    it 'can create a group in a site' do
+      VCR.use_cassette('groups') do
+        group = client.groups.create(name: 'testgroup')
+        expect(group['id']).to be_a_tableau_id
+        expect(group).to eq('id' => group['id'], 'name' => 'testgroup')
+      end
+    end
+  end
+
+  describe '#list' do
+    # http://onlinehelp.tableau.com/v9.0/api/rest_api/en-us/help.htm#REST/rest_api_ref.htm#Query_Groups
+    it 'can list groups in a site' do
+      VCR.use_cassette('groups') do
+        group = client.groups.list.find do |g|
+          g['name'] == 'testgroup'
+        end
+        expect(group['id']).to be_a_tableau_id
```

Comment:
You call these four lines three times. It seems stylistically consistent to pull the first three out into their own function then to remove the duplicate assertions.